src/libs/shared/shared/threading/Utility.cpp

```diff
@@ -166,13 +166,13 @@ std::expected<std::wstring, Result> get_name(auto& thread) {
 	return std::wstring(buffer.data(), buffer.data() + wcslen(buffer.data()));
 #elif defined TARGET_OS_MAC
 	std::array<char, BUFFER_LEN> buffer{};
-	auto res = pthread_getname_np(buffer.data(), buffer.size()); // todo, taking a guess here, can't test
+	auto res = pthread_getname_np(pthread_self(), buffer.data(), buffer.size()); // testing it right now, it needs pthread_self()
```

Comment:
The first argument should be `thread` from the function's arguments. This is a templated function, so the type should match that of `pthread_self()`.

Reply:
It looks like this would be a duplicate of the code for Linux, so it should be possible to just change the defines and remove the MacOS specific section entirely.

Also, if this then works on MacOS, you could remove the MacOS defines from the `get_name(...)` functions that currently print the not implemented warning.